### DISCUSSION TITLE
request provide asinh(x) C function also

### DISCUSSION THREAD

**Discussion by sanjayjain159** [msteveb/jimtcl#298] [Ideas]
Dear Sir,   Kindly provide asinh(x) c function. It is useful in computing log(1+x) when x is small. Or you can provide atanh(x).

**Comment by msteveb** [comment]
Yes, that would be possible, but where to stop. In general I follow the math functions that Tcl provides, although I note that Tcl provides sinh, cosh and tanh which Jim Tcl doesn't, but it doesn't provide any of the inverse functions. Also, what about other functions like gamma, lgamma, bessel functions etc.? Maybe it would be best to make math functions extensible and perhaps have an "advanced math" extension that makes as many of these available as are supported by the C library

**Reply by sanjayjain159** [reply]
Very true. There is no limit to feature addition. I asked for asinh because given sinh and asinh, all hyperbolic functions and their inverses, and also the C functions expm1 and log1p (log(1+x)) can be computed from sinh & asinh. It takes only a short script to obtain these functions from sinh and asinh. But you are rightly concerned about unnecessary bloat.

**Reply by sanjayjain159** [reply]
The Jim Tcl reference manual says the hyperbolic functions (but not their inverses) are provided if math support is enabled.

**Reply by msteveb** [reply]
You're right. My mistake, Jim Tcl does have hyperbolic functions, same as Tcl, just not the inverse. I will look into expanding these, but will probably be a while

**Reply by sanjayjain159** [reply]
Thank you very much.
